src/contexts/ApiPromiseContextProvider.tsx

```diff
@@ -42,29 +42,48 @@ export function ApiPromiseContextProvider(props: ApiRxContextProviderProps): Rea
   const options = { hasher, provider, types };
   const [apiPromise, setApiPromise] = useState<ApiPromise>(new ApiPromise(options));
   const [isReady, setIsReady] = useState(false);
+  const [disconnected, setDisconnected] = useState(false);
 
   useEffect(() => {
     if (isReady) {
+      logger.info(`${contextType} was changed`);
       setIsReady(false);
-      apiPromise.disconnect().then(() => logger.info(`${contextType} Resetting connection`));
+      setApiPromise(new ApiPromise(options));
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [sourceTarget]);
 
   useDidUpdateEffect(() => {
-    ApiPromise.create(options).then((_api) => {
-      setApiPromise(_api);
-    });
-  }, [provider]);
+    if (!isReady && !apiPromise.isConnected) {
+      apiPromise.disconnect().then(() => {
+        setDisconnected(true);
+        logger.info(`${contextType} Resetting connection`);
+      });
+    }
+  }, [isReady]);
```

Comment:
```suggestion
  }, [isReady, apiPromise]);
```

shouldn't this depend on `apiPromise`?